packages/api/tests/endpoints/test-s3credentials.js

```diff
@@ -0,0 +1,284 @@
+'use strict';
+
+/* eslint-disable lodash/prefer-noop */
+const cryptoRandomString = require('crypto-random-string');
+const test = require('ava');
+const sinon = require('sinon');
+const rewire = require('rewire');
+const randomString = () => cryptoRandomString({ length: 6 });
+
+const randomId = (prefix, separator = '-') =>
+  [prefix, randomString()].filter((x) => x).join(separator);
+
+process.env.EARTHDATA_CLIENT_ID = randomId('edlID');
+process.env.EARTHDATA_CLIENT_PASSWORD = randomId('edlPW');
+process.env.DISTRIBUTION_REDIRECT_ENDPOINT = 'http://example.com';
+process.env.DISTRIBUTION_ENDPOINT = `https://${randomId('host')}/${randomId('path')}`;
+process.env.AccessTokensTable = randomId('tokenTable');
+process.env.TOKEN_SECRET = randomId('tokenSecret');
+
+const {
+  s3credentials,
+  buildRoleSessionName,
+  requestTemporaryCredentialsFromNgap,
+  fetchPolicyForUser,
+  parseBucketKey,
+  formatAllowedBucketKeys,
+  configuredForACLCredentials,
+} = require('../../endpoints/s3credentials');
+const index = rewire('../../endpoints/s3credentials.js');
```

Comment:
probably rename that since it's not rewiring the index.js file anymore

Reply:
I just kept it as index because the name of the file "s3credentials" is already used as the name of a function

Reply:
we could rename to something meaningful. also the `.js` extension is not necessary

Reply:
changed.